lib/metasploit/framework/compiler/mingw.rb

```diff
@@ -59,15 +60,54 @@ def build_cmd(src)
           cmd
         end
 
-        def cleanup_files
-          src_file = "#{self.file_name}.c"
-          exe_file = "#{self.file_name}.exe"
 
+        def compile_cpp(src)
+          cmd = build_cpp_cmd(src)
+          print("#{cmd}\n") if show_compile_cmd
+          stdin_err, status = Open3.capture2e(cmd)
+          stdin_err
+        end
+
+        def compile_cpp_files(files)
+          cmd = build_cpp_files_cmd(files)
+          print("#{cmd}\n") if show_compile_cmd
+          stdin_err, status = Open3.capture2e(cmd)
+          stdin_err
+        end
+
+        def build_cpp_cmd(src)
+          src_file = Tempfile.create(['scexec', ".cpp"])
```

Comment:
```suggestion
          src_file = Tempfile.create(['scexec', '.cpp'])
```

Single quotes as double quotes are needed here to keep in line with RuboCop standards. Also do we have to use the name `srcexec` or can this name be randomized (assuming this might be called on remote targets here, if its local only then a hardcoded string is fine but not preferential).

Reply:
Yep makes sense I didn't have strong feelings about it if it was local file, more if it was being done on a remote system. Locally, I don't think anyone really cares too much what you name it so long as its not swearing 😄